components/cam/src/chemistry/bulk_aero/aero_model.F90

```diff
@@ -600,12 +608,34 @@ subroutine aero_model_wetdep( dt, dlf, dlf2, cmfmc2, state, sh_e_ed_ratio,    &
 
     type(wetdep_inputs_t) :: dep_inputs  ! obj that contains inputs to wetdepa routine
 
+    integer           :: nmodes
+    logical           :: clim_modal_aero ! true when radiatively constituents present (nmodes>0)
+    logical           :: prog_modal_aero ! Prognostic modal aerosols present
+
+    ! set flags to check aerosol settings
+    call rad_cnst_get_info(0, nmodes=nmodes)
+    clim_modal_aero = (nmodes > 0)
+    call phys_getopts(prog_modal_aero_out=prog_modal_aero)
```

Comment:
Sorry, I didn't notice it earlier but I think the above three lines belongs to the init routine. These logicals will not change during the simulation.

I can imagine how this works in case we have chem set to `none` (when we are using prescribed aerosols). I do not know how this plays out if somebody tries to revive BAM (I don't think anybody is working on it though). Unless there is a special case, I think this code is not even compiled when `prog_model_aero` or `clim_modal_aero` is true, so I wonder if we need these checks. May be we can try removing these checks and see if we have any unexpected failures. This is one way to just make sure that everything we care about is still working as expected.

Reply:
ok I can do that. The way I've implemented things here should preserve the previous behavior, even for BAM, so I don't think there's anything to worry about there. I'm still a little confused why these flags are used this way though. I would think that this all would have been built to specifically check whether the aerosol option was "bam" or "prescribed" or whatever, but these flags make it hard to understand what's going on.

Reply:
I agree. I wish we had clear logicals for each configurable option. Model is configurable in multiple ways so it is hard play out all the cases. We mostly care about cases for which we have the nightly tests. If all the tests are passing, we generally consider it safe to integrate. 

As far as I know, I do not think we need to checks for `prog_model_aero` and `clim_modal_aero` here but it doesn't hurt to add these.